python/dgl/data/qm9_edge.py

```diff
@@ -0,0 +1,402 @@
+""" 
+    QM9 dataset for graph property prediction (regression).
+    When building this class, we partially refer to the implementation from https://github.com/Jack-XHP/DGL_QM9EDGE  
+"""
+
+import os
+import numpy as np
+
+from .dgl_dataset import DGLDataset
+from .utils import download, extract_archive, _get_dgl_url
+from ..convert import graph as dgl_graph
+from .. import backend as F
+
+try:
+    import rdkit
+    from rdkit import Chem
+    from rdkit.Chem.rdchem import HybridizationType
+    from rdkit.Chem.rdchem import BondType as BT
+    from rdkit import RDLogger
+    RDLogger.DisableLog('rdApp.*')
+except ImportError:
+    rdkit = None
+
+HAR2EV = 27.2113825435      # 1 Hartree = 27.2114 eV 
+KCALMOL2EV = 0.04336414     # 1 kcal/mol = 0.043363 eV
+conversion = F.tensor([
+    1., 1., HAR2EV, HAR2EV, HAR2EV, 1., HAR2EV, HAR2EV, HAR2EV, HAR2EV, HAR2EV,
+    1., KCALMOL2EV, KCALMOL2EV, KCALMOL2EV, KCALMOL2EV, 1., 1., 1.
+])
+
+
+class QM9EdgeDataset(DGLDataset):
+    r"""QM9Edge dataset for graph property prediction (regression)
+    This dataset consists of 130,831 molecules with 19 regression targets.
+    Nodes correspond to atoms and edges correspond to bonds.
+    
+    This dataset differs from :class:`~dgl.data.QM9Dataset` in the following points:
+        1. It includes the bonds in a molecule in the edges of the corresponding graph while the edges in :class:`~dgl.data.QM9Dataset` are purely distance-based.
+        2. It provides edge features, and node features in addition to the atom's locations and atomic number.
+        3. The number of regression targets is expanded from 13 to 19 (in :class:`~dgl.data.QM9Dataset`, it's 13).
+
+    Reference: `"MoleculeNet: A Benchmark for Molecular Machine Learning" <https://arxiv.org/abs/1703.00564>`_
+               Atom features come from `"Neural Message Passing for Quantum Chemistry" <https://arxiv.org/abs/1704.01212>`_
```

Comment:
List references with bullet points as in "Statistics", "Node attributes" and "Edge attributes"